core/cmd/remote_client_test.go

```diff
@@ -311,19 +345,20 @@ func TestClient_RemoteBuildCompatibility(t *testing.T) {
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			enteredStrings := []string{cltest.APIEmail, cltest.Password}
-			prompter := &cltest.MockCountingPrompter{EnteredStrings: enteredStrings}
+			prompter := &cltest.MockCountingPrompter{T: t, EnteredStrings: enteredStrings}
 			client := app.NewAuthenticatingClient(prompter)
 
-			static.Version = test.cliVersion
-			static.Sha = test.cliSha
 			client.HTTP = &mockHTTPClient{client.HTTP, test.remoteVersion, test.remoteSha}
 
-			set := flag.NewFlagSet("test", 0)
-			set.Bool("bypass-version-check", test.bypassVersionFlag, "")
-			c := cli.NewContext(nil, set, nil)
-			err := client.RemoteLogin(c)
+			err := client.CheckRemoteBuildCompatibility(logger.TestLogger(t), test.bypassVersionFlag, test.cliVersion, test.cliSha)
```

Comment:
Not sure I understand why you changed this. Do multiple concurrent logins cause race problems? If so I think multiple tests in this file do that